url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/utilities/MongoResults.java

```diff
@@ -0,0 +1,390 @@
+/*   
+   Copyright 2011-2013 The Cassandra Consortium (cassandra-fp7.eu)
+
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package eu.cassandra.sim.utilities;
+
+import java.util.HashMap;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.DBObject;
+
+/**
+ * The wrapper class for storing simulation output results in an MongoDB database.
+ *
+ * @author Kyriakos C. Chatzidimitriou (kyrcha [at] iti [dot] gr)
+ * @author Fani A. Tzima (fani [dot] tzima [at] iti [dot] gr)
+ */
+public class MongoResults implements DBResults{	
+	
+	/** The database name. */
+	private String dbname;
+	
+	/**
+	 * Instantiates a new MongoResults object.
+	 *
+	 * @param adbname the target database name
+	 */
+	public MongoResults(String adbname) {
+ 		dbname = adbname;
+	}
+	
+	/* (non-Javadoc)
+	 * @see eu.cassandra.sim.utilities.DBResults#createTablesAndIndexes()
+	 */
+	@Override
+	public void createTablesAndIndexes() {
+		DBObject index = new BasicDBObject("tick", 1);
+		DBConn.getConn(dbname).getCollection(COL_AGGRRESULTS).createIndex(index);
+		index = new BasicDBObject("tick", 1);
+		DBConn.getConn(dbname).getCollection(COL_AGGRRESULTS_HOURLY).createIndex(index);
+		index = new BasicDBObject("inst_id", 1);
+		index.put("tick", 1);
+		DBConn.getConn(dbname).getCollection(COL_INSTRESULTS).createIndex(index);
+		index = new BasicDBObject("inst_id", 1);
+		index.put("tick", 1);
+		DBConn.getConn(dbname).getCollection(COL_INSTRESULTS_HOURLY).createIndex(index);
+	}
+	
+	/* (non-Javadoc)
+	 * @see eu.cassandra.sim.utilities.DBResults#addKPIs(java.lang.String, double, double, double, double, double)
+	 */
+	@Override
+	public void addKPIs(String inst_id, double maxPower, double avgPower, double energy, double cost, double co2) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 44 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 101-120 into a separate method.